src/Plivo/Resource/ResourceInterface.cs

```diff
@@ -208,17 +208,11 @@ public static Dictionary<string, object> CreateData (List<string> propertyInfos,
             var dict = new Dictionary<string, object> ();
             foreach (PropertyInfo pi in data.GetType ().GetProperties ()) {
                 if (propertyInfos.Contains (pi.Name)) {
-                    if(pi.Name == "firstParty" || pi.Name == "secondParty"){
-                        if(pi.GetValue (data) == null){
-                            throw new PlivoValidationException (pi.Name + "can not be null");
-                        }
-                    }
-                    else{
-                        if (string.IsNullOrEmpty (pi.GetValue (data)))
-                            throw new PlivoValidationException (pi.Name + " is mandatory, can not be null or empty");
-                    }
-                    
+                    var propertyValue = pi.GetValue(data);
+                    if (pi.Name != "firstParty" && pi.Name != "secondParty" && string.IsNullOrEmpty(propertyValue)) 
+                            throw new PlivoValidationException(pi.Name + " is mandatory, can not be null or empty");
```

Comment:
what will be pi.name in case if both first and second party are empty

Reply:
API validation will be used in this case.